### ISSUE TITLE
Rest api for concepts doesn't allow pages after 10000

### ISSUE THREAD

**Issue by lambda-ai** [IHTSDO/snowstorm#75]
The /browser/{branch}/concepts has pagination implemented, but fails after the 10000th element (regardless of page size) with the message "Maximum offset + page size is 10,000.".

**Comment by kaicode** [comment]
Hi @lambda-ai, yes we haven't found a practical use case for that endpoint so it's only a very basic version has been implemented. Do you have a good use case for this?

It is a slight concern someone could use that endpoint on the public browser to download the whole of SNOMED CT which we would like to prevent. It is recommended that anyone wanting the whole of SNOMEDCT should download the RF2 distribution package from their National Release Centre, they can always load that into their own application or host their own Snowstorm instance.

**Comment by danka74** [comment]
Hi,

We are using the endpoint for (trying to) implement some of our extension quality checks and some of the work that we currently do using a relational database import of SNOMED CT (e.g. determining conepts to translate). I created a RxJS Observable (I love Observables...) for managing pagination: E.g. https://github.com/danka74/snowstorm-client-test/blob/master/index.ts

Cheers,
Daniel

**Comment by lambda-ai** [comment]
Hi @kaicode,

I'm using it as a quick replacement to get around the deprecation of https://github.com/IHTSDO/rf2-to-json-conversion. So I'm running my own snowstorm instance, load the release files, and then pull them back out to json. (I wouldn't do that on someone else's production server :))

It would be easier to use https://github.com/IHTSDO/snomed-boot, but I found that there not the full extraction is implemented (there's no language and acceptability on descriptions). We could of course parse the RF2 files ourselves, but we prefer to use supported tools for that, exactly for the case when something changes that we weren't aware of.

In future we may well run the snowstorm docker as part of our release package, but in the short term that feels a bit too dangerous, and customers are waiting for the update. We didn't notice the deprecation of https://github.com/IHTSDO/rf2-to-json-conversion until we had to (urgently) upgrade the snomed release.

Regards,
-Mathias

**Comment by kaicode** [comment]
@lambda-ai I understand your use case. I will fix this for you today in the develop branch.

**Comment by lambda-ai** [comment]
Thanks a lot!

**Comment by kaicode** [comment]
Hi @lambda-ai,

I've just pushed an implementation to the develop branch for unlimited pagination of the `GET /browser/{branch}/concepts` endpoint. 

It uses searchAfter because Elasticsearch does not support offset greater than 10K. I've updated the swagger API notes for that endpoint to explain how to use searchAfter.

I've disabled this feature by default to prevent so people will need to enable it on their own Snowstorm deployments by setting snowstorm.rest-api.allowUnlimitedConceptPagination=true. 
See the [configuration-guide](https://github.com/IHTSDO/snowstorm/blob/master/docs/configuration-guide.md) has options for setting config properties.

We will be cutting a release later this week.

Kind regards,
Kai
